vocabularies/Common.json

```diff
@@ -1288,6 +1288,51 @@
       "@Common.Experimental": true,
       "@Core.IsURL": true,
       "@Core.Description": "Base URL for WebSocket connections"
+    },
+    "RecommendationsAction": {
+      "$Kind": "Term",
+      "$AppliesTo": ["EntityType"],
+      "@Common.Experimental": true,
+      "@Core.Description": "Name of an action (or function) import that retrieves AI-based recommendations (see [template](#Template_RecommendationsAction)) for an entity of the annotated type"
+    },
+    "Template_RecommendationsAction": [
+      {
+        "$Kind": "Action",
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for actions that retrieve AI-based recommendations for an entity and its related entities",
+        "@Core.LongDescription": "The template action itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "EntityKeys",
+            "$Type": "Edm.ComplexType",
+            "@Core.Description": "Nested structure containing the keys of entities for which recommendations shall be retrieved",
+            "@Core.LongDescription": "The nested complex type of this parameter has the same structure as the entity type bearing the [`GetRecommendationsAction`](#GetRecommendationsAction) annotation\n            and its related entity types, but contains only their key properties.\n            If this parameter can be included in a URL, a function can be used instead of an action."
+          }
+        ],
+        "$ReturnType": { "$Collection": true, "$Type": "Common.PropertyRecommendationType" }
+      }
+    ],
+    "PropertyRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "target": { "@Core.Description": "Target for this recommendation, like the `target` in [`Messages`](#Messages)" },
```

Comment:
Is it really sufficient to refer to the target used in the message property? The message property is part of the entity data which the recommendation is not.
Also as the recommendations for several entities may be requested, the key predicates have to be part of the target. So e.g. when fetching recommendations for sales orders 1, 2 and 3, the targets would need to be something like (id=1)/soldToParty?

Reply:
Or would the target be relative to the request URL? This does not seem to be helpful. So the targets are relative to the entity set for which the action is provided. Only that the action is annotated at the entity type which might be used for several entity sets (in theory).
An example for targeting a property of a related entity would not hurt either.

Reply:
We can define the target to be relative to the entity container, if that is preferred.

Reply:
Not sure whether this is helpful for FE. I guess relative to the entityset should be good.